src/server/session.ts

```diff
@@ -1895,14 +1900,27 @@ namespace ts.server {
 
         private getFullNavigateToItems(args: protocol.NavtoRequestArgs): readonly NavigateToItem[] {
             const { currentFileOnly, searchValue, maxResultCount } = args;
+            if (!args.file) {
+                const items: NavigateToItem[] = [];
+                this.projectService.forEachEnabledProject(project => {
+                    this.projectService.loadAncestorProjectTree();
+                    for (const item of project.getLanguageService().getNavigateToItems(searchValue, maxResultCount, /*filename*/ undefined, /*excludeDts*/ project.isNonTsProject())) {
+                        if (!contains(items, item, navigateToItemIsEqualTo)) {
+                            items.push(item);
+                        }
+                    }
+                });
+                return items;
+            }
+            const fileArgs = args as protocol.FileRequestArgs;

```

Comment:
This seems really odd to me - you've changed the args type to explicitly not be a subtype of `FileRequestArgs` but, knowing that a consumer may still be using it as such and providing a project name, you're casting it back. Why not just leave the type as is if that's the case?

Reply:
Ah - I see, it's because you want the file to be optional. In that case, please keep `projectFileName` in the type?

Reply:
Unless I'm misreading, it was already re-declared on the type.

Reply:
Good ol' diff view, not showing me what you saw. Never mind. :)